reco_utils/dataset/pandas_df_utils.py

```diff
@@ -255,3 +255,39 @@ def negative_feedback_sampler(
     )
 
     return df_sample
+
+
+def has_same_base_dtype(df_1, df_2, columns=None):
+    """Check if specified columns have the same base dtypes
+
+    Args:
+        df_1 (pd.DataFrame): first dataframe
+        df_2 (pd.DataFrame): second dataframe
+        columns (list(str)): columns to check, None checks all columns
+    Returns:
+        bool: True if dataframe columns have the same base dtype
+    """
+
+    result = True
+    for c in columns:
+        if df_1[c].dtype.base != df_2[c].dtype.base:
+            print('Columns {} do not have the same base dtype'.format(c))
+            result = False
```

Comment:
Maybe simply return False in the loop or break, it seems that going through all the columns is unecessary

Reply:
Or compare the results of `df.dtypes.values` with `numpy.all()`.

Reply:
You have to make sure that both dataframes columns are ordered the same way if you do that @anargyri